src/Kernel/IQSharpEngine.cs

```diff
@@ -342,7 +342,31 @@ public override async Task<ExecutionResult> ExecuteMundane(string input, IChanne
                 }
                 catch (CompilationErrorsException c)
                 {
-                    foreach (var m in c.Errors) channel.Stderr(m);
+                    // Check if the user likely tried to execute a magic
+                    // command and try to give a more helpful message in that
+                    // case.
+                    if (input.TrimStart().StartsWith("%") && input.Split("\n").Length == 1)
+                    {
+                        var attemptedMagic = input.Split(" ", 2)[0];
+                        var msg = $"No such magic command {attemptedMagic}.";
+                        if (MagicResolver is MagicSymbolResolver iqsResolver)
+                        {
+                            var similarMagic = iqsResolver
+                                .FindAllMagicSymbols()
+                                .Select(symbol =>
+                                    (symbol.Name, symbol.Name.EditDistanceFrom(attemptedMagic))
+                                )
+                                .OrderBy(pair => pair.Item2)
+                                .Take(3)
+                                .Select(symbol => symbol.Name);
+                            msg += $"\nPossibly similar magic commands:\n\t{string.Join(", ", similarMagic)}";
+                        }
+                        channel.Stderr(msg + "\nTo get a list of all available magic commands, run %lsmagic, or visit https://docs.microsoft.com/qsharp/api/iqsharp-magic/.");
```

Comment:
I think it we output the magics we found as a list, they will actually show like:
* %magic1
* %magic2
* %magic3
which will probably look a little nicer.

Reply:
Good idea, let me go add that.